src/supervisor.ts

```diff
@@ -37,12 +21,11 @@ export class DenoHttpSupervisor {
         return Response.json({ error: "Not found" }, { status: 404 });
       }
 
-      if (moduleName in this.#workerPool.activeWorkers) {
-        const worker = this.#workerPool.activeWorkers[moduleName];
-        return worker.run(req, module);
+      const worker = this.#workers[moduleName] ?? await Worker.create();
+      this.#workers[moduleName] = worker;
+      if (!await worker.healthCheck()) {
+        return Response.json({ error: "Health Check Failed" }, { status: 500 });
       }
```

Comment:
We probably need to do something if the health check is failing. One option would be to remove the worker from the supervisor and shut it down. 

```
      if (!(await worker.healthCheck())) {
        delete this.#workers[moduleName];
        worker.shutdown()
        return Response.json({ error: "Health Check Failed" }, { status: 500 });
      }
```

Reply:
good call - bringing that in my next commit.